urld58fd5e7a9_longrm_weibo_robot_tgz-pJ8-weibo4j_org_json_TestJ8/src/weibo4j/Weibo.java

```diff
@@ -0,0 +1,3668 @@
+
+	/**
+	 * Obtain the ListObject feed list
+	 * @param uid		User ID or screen_name
+	 * @param listId	The ID or slug of ListObject
+	 * @param auth		if true, the request will be sent with BASIC authentication header
+	 * @return
+	 * @throws WeiboException
+	 */
+	public List<Status> getListStatuses(String uid, String listId, boolean auth) throws WeiboException {
+		StringBuilder sb = new StringBuilder();
+		sb.append(getBaseURL()).append(uid).append("/lists/").append(listId).append("/statuses.xml").append("?source=").append(CONSUMER_KEY);
+		String httpMethod = "GET";
+		String url = sb.toString();
+		//
+		return Status.constructStatuses(http.httpRequest(url, null, auth, httpMethod), this);
+	}
+
+
+	/**
+	 * Obtain ListObject member list
+	 * @param uid		User ID or screen_name
+	 * @param listId	The ID or slug of ListObject
+	 * @param auth		if true, the request will be sent with BASIC authentication header
+	 * @return
+	 * @throws WeiboException
+	 */
+	public UserWapper getListMembers(String uid, String listId, boolean auth) throws WeiboException {
+		StringBuilder sb = new StringBuilder();
+		sb.append(getBaseURL()).append(uid).append("/").append(listId).append("/members.xml").append("?source=").append(CONSUMER_KEY);
+		String httpMethod = "GET";
+		String url = sb.toString();
+		//
+		return User.constructWapperUsers(http.httpRequest(url, null, auth, httpMethod), this);
+	}
+
+	/**
+	 * Obtain ListObject subscribe user's list 
+	 * @param uid		User ID or screen_name
+	 * @param listId	The ID or slug of ListObject
+	 * @param auth		if true, the request will be sent with BASIC authentication header
+	 * @return
+	 * @throws WeiboException
+	 */
+	public UserWapper getListSubscribers(String uid, String listId, boolean auth) throws WeiboException {
+		StringBuilder sb = new StringBuilder();
+		sb.append(getBaseURL()).append(uid).append("/").append(listId).append("/subscribers.xml").append("?source=").append(CONSUMER_KEY);
+		String httpMethod = "GET";
+		String url = sb.toString();
+		//
+		return User.constructWapperUsers(http.httpRequest(url, null, auth, httpMethod), this);
+	}
+
+
+
+	/**
+	 * confirm list member
+	 * @param uid		User ID or screen_name
+	 * @param listId	The ID or slug of ListObject
+	 * @param targetUid	Target user ID or screen_name
+	 * @param auth		if true, the request will be sent with BASIC authentication header
+	 * @return
+	 * @throws WeiboException
+	 */
+	public boolean isListMember(String uid, String listId, String targetUid, boolean auth)
+			throws WeiboException {
+		StringBuilder sb = new StringBuilder();
+		sb.append(getBaseURL()).append(uid).append("/").append(listId).append("/members/").append(targetUid)
+				.append(".xml").append("?source=").append(CONSUMER_KEY);
+		String url = sb.toString();
+		//
+		String httpMethod = "GET";
+		//
+		Document doc = http.httpRequest(url, null, auth, httpMethod).asDocument();
+		Element root = doc.getDocumentElement();
+		return "true".equals(root.getTextContent());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 2176:2185, 2199:2208.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.